python/Ganga/new_tests/GPI/GangaUnitTest.py

```diff
@@ -76,6 +76,27 @@ def start_ganga(gangadir_for_test='$HOME/gangadir_testing', extra_opts=[]):
         logger.info("Initializing")
         Ganga.Runtime._prog.initEnvironment(opt_rexec=False)
     else:
+        from Ganga.Runtime.GPIexport import exportToGPI
+
+        from Ganga.Runtime import Repository_runtime
+        # boostrap the repositories and connect to them
+        for n, k, d in Repository_runtime.bootstrap():
+            # make all repository proxies visible in GPI
+            exportToGPI(n, k, 'Objects', d)
+
+        # JobTree
+        from Ganga.Core.GangaRepository import getRegistry
+        jobtree = getRegistry("jobs").getJobTree()
+        exportToGPI('jobtree', jobtree, 'Objects', 'Logical tree view of the jobs')
+
+        # ShareRef
+        shareref = getRegistry("prep").getShareRef()
+        exportToGPI('shareref', shareref, 'Objects', 'Mechanism for tracking use of shared directory resources')
+
+        # The queues are shut down by the atexit handlers so we need to start them here
+        from Ganga.Core.GangaThread.WorkerThreads import startUpQueues
+        startUpQueues()
+
```

Comment:
This duplicates quite a lot of code from bootstrap.py. Is there any way to avoid this?